src/Phinx/Console/Command/SeedRun.php

```diff
@@ -57,45 +57,16 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $this->bootstrap($input, $output);
 
         $seedSet = $input->getOption('seed');
-        $environment = $input->getOption('environment');
-
+        $environment = $this->getEnvironment($input, $output);
         if ($environment === null) {
-            $environment = $this->getConfig()->getDefaultEnvironment();
-            $output->writeln('<comment>warning</comment> no environment specified, defaulting to: ' . $environment);
-        } else {
-            $output->writeln('<info>using environment</info> ' . $environment);
-        }
-
-        if (!$this->getConfig()->hasEnvironment($environment)) {
-            $output->writeln(sprintf('<error>The environment "%s" does not exist</error>', $environment));
-
             return self::CODE_ERROR;
         }
 
         $envOptions = $this->getConfig()->getEnvironment($environment);
-        if (isset($envOptions['adapter'])) {
-            $output->writeln('<info>using adapter</info> ' . $envOptions['adapter']);
-        }
-
-        if (isset($envOptions['wrapper'])) {
-            $output->writeln('<info>using wrapper</info> ' . $envOptions['wrapper']);
-        }
-
-        if (isset($envOptions['name'])) {
```

Comment:
The printing out of environment information doesn't seem to fix it. I assume removing this check is the actual fix.

Reply:
The printing out of information is where it was crashing, because name was not set. I could have simply added the check to all three files, but that's not coding. Why I condensed it while making my sqlite fix (with the exact lines posted above).

Reply:
Just linking me that code didn't explain anything. You're defending a decision that I don't even know happened.

Reply:
The problem is that having an environment of:

```
        adapter: sqlite
        memory: true
```

gives you:

```
$envOptions = [
    'adapter' => 'sqlite',
    'memory' => true
];
```

which when it hit:
```
        if (isset($envOptions['name'])) {
            $output->writeln('<info>using database</info> ' . $envOptions['name']);
        } else {
            $output->writeln('<error>Could not determine database name! Please specify a database name in your config file.</error>');

            return self::CODE_ERROR;
        }
```
would fail the if statement, enter the else and return an error because name was not set. When I consolidated these if checks from these three different commands, I added the special check of:

```php
        if (
            isset($envOptions['adapter'])
            && $envOptions['adapter'] === 'sqlite'
            && isset($envOptions['memory'])
            && $envOptions['memory'] === true
        ) {
```

on that if/else to catch this case, and so now the above configuration works fine and without error. This gets the [second sqlite configuration example in the docs](https://book.cakephp.org/phinx/0/en/configuration.html#id1) to work.